src/Calendar.php

```diff
@@ -398,16 +398,36 @@ public function isAWorkingHour($time)
      * if delay >= DAY_TIMESTAMP : work in days
      * else work in minutes
      *
-     * @param datetime $start               begin
+     * @param string   $start               begin
      * @param integer  $delay               delay to add (in seconds)
      * @param integer  $additional_delay    delay to add (default 0)
      * @param boolean  $work_in_days        force working in days (false by default)
      * @param boolean  $end_of_working_day  end of working day (false by default)
+     * @param boolean  $force               force computation even with no delay
      *
      * @return boolean|string end date
      **/
-    public function computeEndDate($start, $delay, $additional_delay = 0, $work_in_days = false, $end_of_working_day = false)
-    {
+    public function computeEndDate(
+        $start,
+        $delay,
+        $additional_delay = 0,
+        $work_in_days = false,
+        $end_of_working_day = false,
+        $force = false
+    ) {
+        // TODO 10.1: parameter $work_in_day make calculation for duration exprimed
+        // in days (e.g "+ 5 days") but we don't have anything for month.
+        // +1 month will push the date 30 working day when it should get the next
+        // valid calendar date at least one month away from the starting date.
+
+        // Prevent useless computation
+        // This also prevent a bug when using $end_of_working_day and an escalation level
+        // where the date will be pushed to the next day despite $delay and $additional_delay
+        // being both set to 0
+        // It seems that through the $force parameter
+        if ($delay == 0 && $additional_delay == 0 && !$end_of_working_day && !$force) {
+            return $start;
+        }
```

Comment:
IMHO, this kind of code that will be unmaintainable, as it is pretty hard to understand why it is here.